src/functions.js

```diff
@@ -126,23 +126,71 @@ let _ = Mavo.Functions = {
 		}
 	},
 
-	url: (id, url = location) => {
+	/**
+	 * Get the page URL (if called with no params), or a parameter from it.
+	 * @param {string} id Parameter name
+	 * @param {string | URL} [url] URL to get parameter from. Defaults to current page URL
+	 * @param {object} [...options] One or more options objects
+	 * @param { "query" | "path" } [options.type] Type of parameter to get.
+	 * 		"query" for query string type URLs (?foo=value)
+	 * 		"path" for path type URLs (/foo/value)
+	 * 		If not specified, it will search for both, with query string parameters taking precedence
+	 * @param {boolean} [options.case_sensitive=false] Whether to do a case sensitive search
+	 * @param {boolean} [options.multiple=false] Whether to return multiple values if there are multiple parameters with the same name
+	 * @returns {string | null}
+	 */
+	url: (id, url = location, ...options) => {
```

Comment:
Why do we need the `url` parameter here? I [searched through our codebase](https://github.com/search?q=repo:mavoweb/mavo%20.url(&type=code) and found no places where we use it. Moreover, [we didn't document](https://mavo.io/docs/functions/#url) that `url()` has an optional second parameter. What am I missing?

If we leave it as it is, we should provide a value for it every time we need to perform, e.g., a case-insensitive search, like so: `url("foo", location, case_insensitive: true)`.

Reply:
I think I get why we might need it. But what if we make `url` one of the options, too, so we don't have to provide it all the time? Then we can analyze any URLs (not only the one of the current page) with the `url()` function.

Reply:
Fair enough, it's totally possible to want to specify options, and you shouldn't also need to specify the URL. Can you make the change?

Reply:
Already did; just waited for your answer to push. :)

Reply:
Looks good!

Reply:
Nice!